dandiapi/api/models/asset_paths.py

```diff
@@ -0,0 +1,47 @@
+from __future__ import annotations
+
+from django.db import models
+from django.db.models import Q
+from django_extensions.db.models import TimeStampedModel
+
+from dandiapi.api.models.asset import Asset
+from dandiapi.api.models.version import Version
+
+
+class AssetPath(TimeStampedModel):
+    path = models.CharField(max_length=512)
+
+    # TODO: Determine what should be done on delete
+    asset = models.ForeignKey(
+        Asset, null=True, blank=True, related_name='leaf_paths', on_delete=models.CASCADE
```

Comment:
I don't have any better suggestions, but it is counter intuitive that asset paths might not have an asset. #namingthingsishard